### PR TITLE
Make google auth optional on server side

### PR DESCRIPTION
**Pull request by charlesBochet**
In this PR, I'm making the following changes:
- adding a `AUTH_GOOGLE_ENABLED` boolean to .env. If set to true, `AUTH_GOOGLE_CLIENT_ID`, `AUTH_GOOGLE_CLIENT_SECRET` and `AUTH_GOOGLE_CALLBACK_URL` will be required. If missing or set to any other value, these won't be required
- if `AUTH_GOOGLE_ENABLED` is set to false, `GoogleStrategy` is set with placeholder values for credentials and google auth controller routes are disabled through a `GoogleProviderEnabledGuard`

Additionally:
- rename storage env variables to be less confusing: if `STORAGE_TYPE=local`, `STORAGE_LOCAL_PATH` is required / if `STORAGE_TYPE=S3`, `STORAGE_S3_REGION` and `STORAGE_S3_NAME` are required

### CODE REVIEW

File: server/src/core/auth/strategies/google.auth.strategy.ts [hunk 1]
```diff
@@ -16,10 +16,17 @@ export type GoogleRequest = Request & {
 @Injectable()
 export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
   constructor(environmentService: EnvironmentService) {
+    const isAuthGoogleEnabled = environmentService.getAuthGoogleEnabled();
     super({
-      clientID: environmentService.getAuthGoogleClientId(),
-      clientSecret: environmentService.getAuthGoogleClientSecret(),
-      callbackURL: environmentService.getAuthGoogleCallbackUrl(),
+      clientID: isAuthGoogleEnabled
```
Comment (charlesBochet) [inline server/src/core/auth/strategies/google.auth.strategy.ts:21]: @magrinj @emilienchvt not very proud of these disabled but `super() `has to be called.
Another solution would be to not register the strategy using DynamicModules

File: server/src/integrations/environment/environment.validation.ts [hunk 2]
```diff
@@ -38,34 +39,44 @@ export class EnvironmentVariables {
...
   @IsString()
-  STORAGE_LOCATION: string;
+  @ValidateIf((env) => env.STORAGE_TYPE === StorageType.Local)
+  STORAGE_LOCAL_PATH?: string;
 }
 
 export function validate(config: Record<string, unknown>) {
   const validatedConfig = plainToClass(EnvironmentVariables, config, {
-    enableImplicitConversion: true,
+    enableImplicitConversion: false,
```
Comment (charlesBochet) [inline server/src/integrations/environment/environment.validation.ts:79]: I have disabled this because it was casting ENV_VARIABLE=false to TRUE...

File: server/src/integrations/integrations.module.ts [hunk 3]
```diff
@@ -16,23 +16,16 @@ import { assert } from 'src/utils/assert';
...
-    return { bucketName };
-  }
-
-  assert(region, 'S3 region is not defined');
+  const bucketName = environmentService.getStorageS3Name();
+  const region = environmentService.getStorageS3Region();
 
   return {
-    bucketName,
+    bucketName: bucketName ?? '',
```
Comment (charlesBochet) [inline server/src/integrations/integrations.module.ts:23]: I'm only moving the problem to here. I believe the S3Module should not be instanciated if S3 is not configured in env variables. I would encapsulated LocalDriver and S3Driver behind one common FileStorage Interface.
To be discussed and treated in another PR!

File: server/src/integrations/environment/environment.validation.ts [hunk 4]
```diff
@@ -38,34 +39,44 @@ export class EnvironmentVariables {
   @IsUrl({ require_tld: false })
   FRONT_AUTH_CALLBACK_URL: string;
 
-  @IsString()
+  @Transform(({ value }) => value.toLowerCase() === 'true')
```
Comment (charlesBochet) [inline server/src/integrations/environment/environment.validation.ts:42]: Ideally, I would like the environment validation to throw if the provided value is not either `'true'`/`'TRUE'`/`'1'` or `'false'`/`'FALSE'`/`'0'` and then convert it to a boolean. I'm unsure how to do that properly. 
I'm thinking about a pre-process of environment variables maybe a little bit more sophisticated than `@Transform` or `enableImplicitConversion`

File: server/src/integrations/environment/environment.validation.ts (hunk 2 above)
Comment (charlesBochet) [reply server/src/integrations/environment/environment.validation.ts:79]: see my other comment below

Comment (emilienchvt) [reply server/src/integrations/environment/environment.validation.ts:79]: Indeed, maybe some ideas here: https://github.com/typestack/class-transformer/issues/550

File: server/src/core/auth/strategies/google.auth.strategy.ts (hunk 1 above)
Comment (emilienchvt) [reply server/src/core/auth/strategies/google.auth.strategy.ts:21]: Not extremely better, but disabledClientId, disabledClientSecret and disabledCallbackUrl might be less generic

File: server/src/integrations/integrations.module.ts (hunk 3 above)
Comment (emilienchvt) [reply server/src/integrations/integrations.module.ts:22]: Should we track this as a new task ?

Comment (charlesBochet) [reply server/src/integrations/integrations.module.ts:22]: Yes, I'll discuss it with Jeremy tomorrow and create a task based on discussion!

File: server/src/integrations/environment/environment.validation.ts (hunk 2 above)
Comment (charlesBochet) [reply server/src/integrations/environment/environment.validation.ts:79]: nice!

**Review by emilienchvt — APPROVED** [review]

File: server/src/integrations/integrations.module.ts (hunk 3 above)
Comment (charlesBochet) [reply server/src/integrations/integrations.module.ts:22]: Created here: https://github.com/twentyhq/twenty/issues/510 we can discuss it tomorrow

File: server/src/integrations/environment/environment.validation.ts (hunk 2 above)
Comment (magrinj) [reply server/src/integrations/environment/environment.validation.ts:79]: I prefer to use `Transform` instead of disabling this, because it's going to stop inferring types on all other env variables
